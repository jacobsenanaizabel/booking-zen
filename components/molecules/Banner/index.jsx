import React from 'react';
import PropTypes from 'prop-types';
import { Title } from '@components/atoms';
import { Search } from '@components/organisms';
import banner from '@public/images/banner.png'

import {
  Container,
  ImgBanner,
  ContainerImage,
  ContainerContent
} from './styles';

const Banner = ({searchFreelancer}) => (
  <>
    <Container>
      <ContainerContent>
        <Title text="Find your expert &#127881;" />
        <Search searchFreelancer={searchFreelancer}/>
      </ContainerContent>
      <ContainerImage>
        <ImgBanner 
          alt="Banner"
          quality={100}
          layout='responsive'
          src={banner}/>
      </ContainerImage>
    </Container>
  </>
);

Banner.propTypes = {
  searchFreelancer: PropTypes.func
};


export default Banner;
