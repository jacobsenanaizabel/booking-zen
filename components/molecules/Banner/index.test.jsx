import React from 'react';
import { render } from "@testing-library/react";
import {Banner} from '@components/molecules/index';

jest.mock('@public/images/banner.png', () => {
  return {
    default:  "/",
  }
});

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props) => {
    return <img {...props} />
  },
}))

describe('Banner tests', () => {

  it('should match snapshot with default Banner', () => {
    
    const { asFragment } = render(<Banner />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Banner with searchFreelance', () => {
    
    const functionSearch = jest.fn();

    const { asFragment } = render(<Banner searchFreelancer={functionSearch}/>);
    expect(asFragment()).toMatchSnapshot();
  });

});