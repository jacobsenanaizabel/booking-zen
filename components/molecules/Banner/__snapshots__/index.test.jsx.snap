// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner tests should match snapshot with default Banner 1`] = `
<DocumentFragment>
  <section
    class="styles__Container-sc-12btswz-0 bVjqLC"
  >
    <div
      class="styles__ContainerContent-sc-12btswz-2 iKmYnb"
    >
      <div
        class="styles__ContainerTitle-sc-17f0diz-1 kMIxue"
      >
        <h1
          class="styles__Heading-sc-17f0diz-0 iYJgeC"
        >
          Find your expert ðŸŽ‰
        </h1>
      </div>
      <div
        class="styles__Container-sc-v1l7xw-0 kMwZu"
      >
        <input
          class="styles__InputStyled-sc-1i43h77-0 fULZLo"
          data-testid="input"
          name="search"
          placeholder="Find an expert by specialty or name"
          type="text"
          value=""
        />
        <div
          class="styles__Container-sc-os8su5-0"
        >
          <button
            class="styles__ButtonStyled-sc-os8su5-1 faeSvG"
            role="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
    <div
      class="styles__ContainerImage-sc-12btswz-3 bOMnKa"
    >
      <img
        alt="Banner"
        class="styles__ImgBanner-sc-12btswz-1 cGcVfH"
        layout="responsive"
        quality="100"
        src="[object Object]"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`Banner tests should match snapshot with default Banner with searchFreelance 1`] = `
<DocumentFragment>
  <section
    class="styles__Container-sc-12btswz-0 bVjqLC"
  >
    <div
      class="styles__ContainerContent-sc-12btswz-2 iKmYnb"
    >
      <div
        class="styles__ContainerTitle-sc-17f0diz-1 kMIxue"
      >
        <h1
          class="styles__Heading-sc-17f0diz-0 iYJgeC"
        >
          Find your expert ðŸŽ‰
        </h1>
      </div>
      <div
        class="styles__Container-sc-v1l7xw-0 kMwZu"
      >
        <input
          class="styles__InputStyled-sc-1i43h77-0 fULZLo"
          data-testid="input"
          name="search"
          placeholder="Find an expert by specialty or name"
          type="text"
          value=""
        />
        <div
          class="styles__Container-sc-os8su5-0"
        >
          <button
            class="styles__ButtonStyled-sc-os8su5-1 faeSvG"
            role="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
    <div
      class="styles__ContainerImage-sc-12btswz-3 bOMnKa"
    >
      <img
        alt="Banner"
        class="styles__ImgBanner-sc-12btswz-1 cGcVfH"
        layout="responsive"
        quality="100"
        src="[object Object]"
      />
    </div>
  </section>
</DocumentFragment>
`;
